karapace/avro_compatibility.py

```diff
@@ -1,10 +1,32 @@
 from avro.schema import (
     ARRAY, ArraySchema, BOOLEAN, BYTES, DOUBLE, ENUM, EnumSchema, Field, FIXED, FixedSchema, FLOAT, INT, LONG, MAP,
-    MapSchema, NamedSchema, NULL, RECORD, RecordSchema, Schema, STRING, UNION, UnionSchema
+    MapSchema, NamedSchema, Names, NULL, RECORD, RecordSchema, Schema, SchemaFromJSONData, STRING, UNION, UnionSchema
 )
 from copy import copy
 from enum import Enum
-from typing import cast, Dict, List, Optional, Set
+from typing import Any, cast, Dict, List, Optional, Set
+
+import json
+
+
+def parse_json_ignore_trailing(s: str) -> Any:
+    """ Compatibility function with Avro which ignores trailing data in JSON
+    strings.
```

Comment:
Out of curiosity: when does this happen?

Reply:
In the test case it would be here: https://github.com/confluentinc/schema-registry/blob/v5.0.3/core/src/test/java/io/confluent/kafka/schemaregistry/avro/AvroCompatibilityTest.java#L69-L75

And it seems the production code also uses the same parsing facilities: https://github.com/confluentinc/schema-registry/blob/v5.0.3/core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java#L821

I did not check, however, if the configuration can be globally changed.